test/jdk/java/io/File/createTempFile/SpecialTempFile.java

```diff
@@ -108,9 +108,19 @@ public static void main(String[] args) throws Exception {
         // Test JDK-8013827
         String[] resvPre = { "LPT1.package.zip", "com7.4.package.zip" };
         String[] resvSuf = { ".temp", ".temp" };
+        String osName = StaticProperty.osName();
+        OSVersion osVersion = OSVersion.current();
+        System.out.println("OS name:    " + osName + "\n" +
+                           "OS version: " + osVersion);
+
+        String[] nameElements = osName.split(" ");
+        int nameVers = Integer.valueOf(nameElements[nameElements.length - 1]);
```

Comment:
That may work with current versions but be aware we sometimes have things like "Server 2012 R2"

Reply:
Thanks. Yes, I looked at various references including [Wikipedia](https://en.wikipedia.org/wiki/List_of_Microsoft_Windows_versions). It could use some refinement, probably, or perhaps the Windows-specific of the test could simply be removed.

Reply:
Also that for the moment is the entire list of [Windows configurations](https://www.oracle.com/java/technologies/javase/products-doc-jdk23certconfig.html) certified for JDK 23.